src/Altinn.App.Core/Features/Validation/Default/LegacyFileAnalyzerValidator.cs

```diff
@@ -0,0 +1,66 @@
+namespace Altinn.App.Core.Features.Validation.Default;
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Altinn.App.Core.Features.FileAnalysis;
+using Altinn.App.Core.Models.Validation;
+using Altinn.Platform.Storage.Interface.Models;
+
+/// <summary>
+/// Implementation of <see cref="IFileUploadValidator"/> that uses the legacy 
+/// <see cref="IFileValidator"/> and <see cref="IFileAnalyser"/>.
+/// </summary>
+public class LegacyFileAnalyzerValidator : IFileUploadValidator
```

Comment:
I think this should be `internal` and `sealed`? This was meant to bridge the migration to the the new interfaces right, so we have some expectations from the behavior here?

Reply:
Should users be able to unregister this? I thought it was needed to still support the file analyzer abstraction

Reply:
I don't see any reason why we should actively prevent users from unregistering anything and replace it with their own interface implementation.

I also can't find any realistic scenario when you'd actually need to unregister this service, but a contrived example might be if you had a badly behaved `IFileAnalyser` in a package that should not be constructed for uploads to all DataTypes, but only the ones where it is actually used. A quick fix would be to replace this implementation that inject `IEnumerable<IFileAnalysr>` with one that fetches from `IServiceProvider` only when it is actually required.

Reply:
It doesn't prevent unregistering though. My impression from previous discussions was that we would be more intentional about type accessibility going forward, and that any modification to public types is a breaking change. That's what I was considering here when reviewing, and so if I consider the tradeoffs between the ergonomics of unregistering and including this as a public API I personally am more worried about adding another public type